code/game/objects/structures/water_structures/sink.dm

```diff
@@ -202,6 +202,24 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/sink, (-14))
 		qdel(O)
 		return
 
+	if(istype(O, /obj/item/ashtray))
```

Comment:
we have 8 different unique istypes for this--we shouldn't add more. pls make this a signal